File: rustnlpjava/src/main/java/dev/gigapixel/rustnlp/BoxErr.java
package dev.gigapixel.rustnlp;

import dev.gigapixel.rustnlp.ntv.*;


import java.lang.foreign.Arena;
import java.lang.foreign.MemorySegment;
import java.lang.foreign.SegmentAllocator;
import java.lang.ref.Cleaner;
import java.util.List;
import static java.lang.foreign.ValueLayout.*;
import java.nio.charset.StandardCharsets;
import java.util.stream.Stream;

public class BoxErr {

    MemorySegment internal;
    Cleaner.Cleanable cleanable;
    SegmentAllocator arena;

    List<Object> selfEdges = List.of();
    

    static class BoxErrCleaner implements Runnable {

        MemorySegment segment;
        BoxErrCleaner(MemorySegment segment) {
            this.segment = segment;
        }

        public void run() {
            rustnlp_h.BoxErr_destroy(this.segment);
        }
    }

    BoxErr() {}
    BoxErr(MemorySegment handle, List<Object> selfEdges) {
        this.internal = handle;
        this.selfEdges = selfEdges;
        

    }
    
    
    public String get() {
        
        try (var arena = Arena.ofConfined()) {
            
            
            var nativeVal = rustnlp_h.BoxErr_get(arena, internal);
            
            var returnVal = SliceUtils.readUtf8(nativeVal);
            return returnVal;
                    
        }
    }
    
}

File: rustnlpkotlin/src/main/kotlin/dev/gigapixel/rustnlp/BoxErr.kt
package dev.gigapixel.rustnlp;
import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure


internal interface BoxErrLib: Library {
    fun BoxErr_destroy(handle: Pointer)
    fun BoxErr_get(handle: Pointer): Slice
}

class BoxErr internal constructor (
    internal val handle: Pointer,
    // These ensure that anything that is borrowed is kept alive and not cleaned
    // up by the garbage collector.
    internal val selfEdges: List<Any>
)  {

    internal class BoxErrCleaner(val handle: Pointer, val lib: BoxErrLib) : Runnable {
        override fun run() {
            lib.BoxErr_destroy(handle)
        }
    }

    companion object {
        internal val libClass: Class<BoxErrLib> = BoxErrLib::class.java
        internal val lib: BoxErrLib = Native.load("rustnlp", libClass)
    }
    
    fun get(): String {
        
        val returnVal = lib.BoxErr_get(handle);
            return PrimitiveArrayTools.getUtf8(returnVal)
    }

}
